### PR TITLE
bdist_msi: add license for msi

### PR DESCRIPTION
**Pull request by ntindle**
# Background

The current bdist_msi command does not support a license confirmation page

# Changes

This PR 
- Adds support for a license confirmation page to the MSI install dialogs

### CODE REVIEW

File: cx_Freeze/command/bdist_msi.py [hunk 1]
```diff
@@ -194,6 +195,7 @@ def add_config(self) -> None:
                 ("A_SET_TARGET_DIR", 'TARGETDIR=""', 401),
                 ("A_SET_REINSTALL_MODE", 'REINSTALLMODE=""', 402),
                 ("SelectDirectoryDlg", "not Installed", 1230),
+                ("LicenseAgreementDlg", "not Installed", 1240),
```
Comment (ntindle) [inline cx_Freeze/command/bdist_msi.py:202]: Note: this doesn't require the dlg be included to be valid

File: cx_Freeze/command/bdist_msi.py [hunk 2]
```diff
@@ -117,6 +117,7 @@ class bdist_msi(Command):
         ("target-name=", None, "name of the file to create"),
         ("target-version=", None, "version of the file to create"),
         ("upgrade-code=", None, "upgrade code to use"),
+        ("license-file=", None, "license file to include in the installer"),
```
Comment (marcelotduarte) [inline cx_Freeze/command/bdist_msi.py]: Does it only accept .rtf and .txt? It is advisable to include the accepted formats.

File: cx_Freeze/command/bdist_msi.py (hunk 1 above)
Comment (marcelotduarte) [reply cx_Freeze/command/bdist_msi.py:202]: Do you mean that, if there is no license file, the dialog will not appear?

File: cx_Freeze/command/bdist_msi.py (hunk 2 above)
Comment (ntindle) [reply cx_Freeze/command/bdist_msi.py]: Only RTF I’ve gotten to render so far

File: cx_Freeze/command/bdist_msi.py (hunk 1 above)
Comment (ntindle) [reply cx_Freeze/command/bdist_msi.py:202]: Exactly, including this line here only tells it the order to render them in, if included but it works fine if the dialog isn’t included

**Comment by marcelotduarte** [comment]
I'm trying to understand why the failures with codecov:
info - 2024-06-21 07:25:41,467 -- The PR is happening in a forked repo. Using tokenless upload.
Error: Codecov token not found. Please provide Codecov token with -t flag.

**Comment by ntindle** [comment]
code cov requires a token to upload coverage. 

The tokens aren’t available on the pull_request push trigger in GitHub actions

not sure how we got around this at AutoGPT. I think installing the code cov app into the org or repo was it though. Can look more if desired

**Comment by marcelotduarte** [comment]
According to the information in the links below, you would not need the token.
https://github.com/marketplace/actions/codecov#breaking-changes
https://docs.codecov.com/docs/frequently-asked-questions#where-is-the-repository-upload-token-found

**Comment by marcelotduarte** [comment]
> not sure how we got around this at AutoGPT. I think installing the code cov app into the org or repo was it though. Can look more if desired

This has already been done and was working well. It had happened to another PR, which is still open and is happening to yours. When I modify, or rebase, they pass.

File: cx_Freeze/command/bdist_msi.py (hunk 2 above)
Comment (ntindle) [reply cx_Freeze/command/bdist_msi.py]: [`e3c500a` (#2472)](https://github.com/marcelotduarte/cx_Freeze/pull/2472/commits/e3c500a76d064fcf5d2d17b58200e2964c6affe0)

Comment (ntindle) [reply cx_Freeze/command/bdist_msi.py]: RTF will be the only supported file. See https://learn.microsoft.com/en-us/windows/win32/msi/scrollabletext-control#control-attributes -> Text for more details

Comment (marcelotduarte) [reply cx_Freeze/command/bdist_msi.py]: Please add this information to the docs, like in the https://cx-freeze--2472.org.readthedocs.build/en/2472/bdist_msi.html#cmdoption-arg-upgrade_code that has a link.
Also fix the option https://cx-freeze--2472.org.readthedocs.build/en/2472/bdist_msi.html#cmdoption-arg-license-file to license_file.

File: doc/src/bdist_msi.rst [hunk 3]
```diff
@@ -62,6 +62,8 @@ command:
        and an executable. Additional allowed keys are `argument` to specify
        the invocation of the executable, `mime` for the extension’s mime type,
        and `context` for the context menu text.
+   * - .. option:: license-file
```
Comment (marcelotduarte) [inline doc/src/bdist_msi.rst]: license_file

File: doc/src/bdist_msi.rst [hunk 4]
```diff
@@ -62,6 +62,8 @@ command:
...
+     - path to an rtf formmated file to be used as the license agreement.
 
 .. versionadded:: 6.7
```
Comment (marcelotduarte) [inline doc/src/bdist_msi.rst:70]: add a versionadded:: 7.2

**Review by marcelotduarte — CHANGES_REQUESTED** [review]

File: doc/src/bdist_msi.rst (hunk 3 above)
Comment (ntindle) [reply doc/src/bdist_msi.rst]: done in [`9c3d270` (#2472)](https://github.com/marcelotduarte/cx_Freeze/pull/2472/commits/9c3d2700529ac1df700b9ea3da782812a625e811)

File: doc/src/bdist_msi.rst (hunk 4 above)
Comment (ntindle) [reply doc/src/bdist_msi.rst:70]: Done in [`9c3d270` (#2472)](https://github.com/marcelotduarte/cx_Freeze/pull/2472/commits/9c3d2700529ac1df700b9ea3da782812a625e811)

**Review by marcelotduarte — APPROVED** [review]

**Comment by marcelotduarte** [comment]
Thanks!
If everything goes well according to my plans, I will release a new version on 7/16, when it will be a holiday in my city.

**Pull request merged**